TheWarWithin/DemonHunterVengeance.lua

```diff
@@ -986,22 +1001,32 @@ end, state )
 
 local sigilList = { "sigil_of_Flame", "sigil_of_misery", "sigil_of_spite", "sigil_of_silence", "sigil_of_chains", "sigil_of_doom" }
 
-local TriggerDemonic = setfenv( function( )
+local TriggerDemonic = setfenv( function()
+
+    local demonicExtension = 7
+
     if buff.metamorphosis.up then
-        buff.metamorphosis.duration = buff.metamorphosis.duration + 7
-        buff.metamorphosis.expires = buff.metamorphosis.expires + 7
-    else
-        applyBuff( "metamorphosis", 7 )
-        if talent.inner_demon.enabled then
-            applyBuff( "inner_demon" )
+        buff.metamorphosis.expires = buff.metamorphosis.expires + demonicExtension
+        if talent.demonic_intensity.enabled and buff.demonsurge_hardcast.up then
+            -- Extend the appropriate buffs
+            local hardcastMetaExpires = buff.metamorphosis.expires
+            applyBuff( "demonsurge_hardcast", buff.metamorphosis.remains )
+            if buff.demonsurge_fel_desolation.up then buff.demonsurge_fel_desolation.expires = hardcastMetaExpires end
+            if buff.demonsurge_consuming_fire.up then buff.demonsurge_consuming_fire.expires = hardcastMetaExpires end
+            if buff.demonsurge_sigil_of_doom.up then buff.demonsurge_sigil_of_doom.expires = hardcastMetaExpires end
         end
```

Comment:
This section also needs to extend `demonsurge_spirit_burst` and `demonsurge_soul_sunder` in the case where you cast Metamorphosis followed immediately by Fel Desolation.

So something like: 
```
if talent.demonsurge.enabled then
    if buff.demonsurge_spirit_burst.up then buff.demonsurge_spirit_burst.expires = hardcastMetaExpires end
    if buff.demonsurge_soul_sunder.up then buff.demonsurge_soul_sunder.expires = hardcastMetaExpires end
end
```